Source/configDsp.cmake

```diff
@@ -47,6 +47,11 @@ if (HELIUM OR MVEF OR SUPPORT)
    target_include_directories(${project} PRIVATE "${DSP}/PrivateInclude")
 endif()
 
+if (MVEI OR HELIUM)
+    # Implicit conversion between vectors of different numbers or types of elements
+    target_compile_options(${project} PRIVATE -flax-vector-conversions)
```

Comment:
Is this option supported on other compilers like, for instance, IAR ?
I would prefer this option to be chosen by the developer rather than imposed by CMSIS-DSP.
Of perhaps we need another cmake option to control this ? Or limit this only to GCC ?

Reply:
> Is this option supported on other compilers like, for instance, IAR ?

Good point. This option is available in Clang (which is probably the reason armclang doesn't have an issue seeing it). I haven't seen it in IAR's manual, so it's probably unavailable.

> I would prefer this option to be chosen by the developer rather than imposed by CMSIS-DSP. Of perhaps we need another cmake option to control this ? Or limit this only to GCC ?

Limiting this to GCC makes sense, I'll add it.

Because some vector code in CMSIS-DSP don't compiler without implicit conversion, I think there's no choice but ensure the conversion is actually enabled, so that the user doesn't need to globally add `-flax-vector-conversions` to their project. Only GCC disables this by default (i.e. encourages code to convert vectors explicitly), whereas Clang-based compilers have it enabled by default.